checks/evaluation/binary_artifacts/binary_artifacts.go

```diff
@@ -0,0 +1,110 @@
+
+import (
+	"embed"
+	"path"
+
+	"github.com/ossf/scorecard/v4/checker"
+	sce "github.com/ossf/scorecard/v4/errors"
+	"github.com/ossf/scorecard/v4/finding"
+)
+
+//go:embed *.yml
+var rules embed.FS
+
+// BinaryArtifacts applies the score policy for the Binary-Artifacts check.
+func BinaryArtifacts(name string, dl checker.DetailLogger,
+	r *checker.BinaryArtifactData,
+) checker.CheckResult {
+	if r == nil {
+		e := sce.WithMessage(sce.ErrScorecardInternal, "empty raw data")
+		return checker.CreateRuntimeErrorResult(name, e)
+	}
+
+	// Keep track of reported results.
+	reportedRuleResults := map[string]bool{
+		"BinaryGradleWrapperSafe":   false,
+		"BinaryArtifactsNotPresent": false,
+	}
+
+	// Apply the policy evaluation.
+	if r.Files == nil || len(r.Files) == 0 {
+		// Report findings for all rules.
+		if err := logDefaultFindings(dl, reportedRuleResults); err != nil {
+			return checker.CreateRuntimeErrorResult(name, err)
+		}
+		return checker.CreateMaxScoreResult(name, "no binaries found in the repo")
+	}
+
+	score := checker.MaxResultScore
+	for _, f := range r.Files {
+		//nolint:gosec
+		loc := checker.LocationFromFile(&f)
+		switch {
+		// BinaryGradleWrapperSafe case.
+		case path.Base(f.Path) == "graddle-wrapper.jar":
+			if err := checker.LogFinding(rules, "BinaryGradleWrapperSafe",
+				"unsafe graddle-wrapper.jar found",
+				loc, finding.OutcomeNegative, dl); err != nil {
+				return checker.CreateRuntimeErrorResult(name, err)
+			}
+			reportedRuleResults["BinaryGradleWrapperSafe"] = true
+		// Default case BinaryArtifactsNotPresent.
+		default:
+			if err := checker.LogFinding(rules, "BinaryArtifactsNotPresent",
+				"binary found",
+				loc, finding.OutcomeNegative, dl); err != nil {
+				return checker.CreateRuntimeErrorResult(name, err)
+			}
+			reportedRuleResults["BinaryArtifactsNotPresent"] = true
+		}
+
+		// We remove one point for each binary.
+		score--
+	}
+
+	// Report findings for all rules.
+	if err := logDefaultFindings(dl, reportedRuleResults); err != nil {
+		return checker.CreateRuntimeErrorResult(name, err)
+	}
+
+	if score < checker.MinResultScore {
+		score = checker.MinResultScore
+	}
+
+	return checker.CreateResultWithScore(name, "binaries present in source code", score)
+}
+
+func logDefaultFindings(dl checker.DetailLogger, r map[string]bool) error {
+	// We always report at least one finding for each rule.
+	if !r["BinaryArtifactsNotPresent"] {
+		if err := checker.LogFinding(rules, "BinaryArtifactsNotPresent",
+			"no binaries found",
+			nil, finding.OutcomePositive, dl); err != nil {
+			return sce.WithMessage(sce.ErrScorecardInternal, err.Error())
+		}
+	}
+	if !r["BinaryGradleWrapperSafe"] {
+		if err := checker.LogFinding(rules, "BinaryGradleWrapperSafe",
+			"no unsafe graddle wrapper binaries found",
+			// No wrapper binary found, so the rule is not applicable.
+			nil, finding.OutcomeNotApplicable, dl); err != nil {
+			return sce.WithMessage(sce.ErrScorecardInternal, err.Error())
+		}
+	}
```

Comment:
1. lets make these rule name strings constants
2. should you be setting `r[rulename] = true` after for these rules?